server/vault.api.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const HomebrewModel = require('./homebrew.model.js').model;
+
+const router = express.Router();
+
+const buildTitleConditions = (title) => {
+    if (!title) return {};
+    return {
+        $text: {
+            $search: title,
+            $caseSensitive: false,
+        },
+    };
+};
+
+const buildAuthorConditions = (author) => {
+    if (!author) return {};
+    return { authors: author };
+};
+
+//"$and": [ {"published": true}, {"$text": { "$search": "titleString", "$caseSensitive": false } }, { "authors" : "authorString"}]
+//is a good example of a query constructed with this function
+
+const handleErrorResponse = (res, error, functionName) => {
+    const status = error.response?.status || 500;
+    const message =
+        status === 503 ? 'Service Unavailable' : 'Internal Server Error';
+
+    console.error(`Error in ${functionName}:`, error);
+
+    return res.status(status).json({
+        errorCode: status.toString(),
+        message: `Error in function ${functionName}: ${message}`,
+    });
+};
+
+const buildBrewsQuery = (legacy, v3) => {
+    //first off, no need to include renderers, if all are valid
+    if (legacy === 'true' && v3 === 'true') return { published: true };
+
+    const renderers = [];
+    if (legacy === 'true') renderers.push('legacy');
+    if (v3 === 'true') renderers.push('V3');
+
+    const brewsQuery = { published: true };
```

Comment:
Might be better to declare this first and return it on line 40 on chance this default grows further

Reply:
good point, also simplifying the syntax.